lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -177,6 +186,101 @@ def __init__(self, *args):
 The consumer charm can then choose to update its configuration in both situations.
 
 
+## LogProxyConsumer Library Usage
+
+Let's say that we have a workload charm that produces logs and we need to send those logs to a
+workload implementing the `loki_push_api` interface, such as `Loki` or `Grafana Agent`.
+
+Adopting this object in a charmed operator consist of two steps:
+
+
+1. Use the `LogProxyConsumer` class by instanting it in the `__init__` method of the charmed
+   operator. There are two ways to get logs in to promtail. You can give it a list of files to read
+   or you can write to it using the syslog protocol.
+
+   For example:
+
+   ```python
+   from charms.loki_k8s.v0.log_proxy import LogProxyConsumer, PromtailDigestError
+
+   ...
+
+       def __init__(self, *args):
+           ...
+           try:
+               self._log_proxy = LogProxyConsumer(
+                   charm=self, log_files=LOG_FILES, container_name=PEER, enable_syslog=True
+               )
+           except PromtailDigestError as e:
+               msg = str(e)
+               logger.error(msg)
+               self.unit.status = BlockedStatus(msg)
```

Comment:
This is IMO not a good pattern: people will copy this, and unknowingly override their status with something else in the first hook that runs. @simskij @jnsgruk do we have better options?

Reply:
I think the pattern we've discussed using until we come up with something better is to have the lib emit some kind of error event, and have the charm author listen to that and decide themselves whether to block or not.

Reply:
I think the pattern we've discussed using until we come up with something better is to have the lib emit some kind of error event, and have the charm author listen to that and decide themselves whether to block or not.

Reply:
This is a bug in the merge I think. We remove these some weeks ago.

Reply:
Let's follow this in issue #69